rust/types/src/metadata.rs

```diff
@@ -679,30 +635,80 @@ impl TryFrom<chroma_proto::WhereDocument> for WhereDocument {
                     Ok(operator) => operator,
                     Err(_) => return Err(WhereConversionError::InvalidWhereChildren),
                 };
-                let children = WhereDocumentChildren {
+                let children = WhereChildren {
                     children: proto_children
                         .children
                         .into_iter()
                         .map(|child| child.try_into())
-                        .collect::<Result<Vec<WhereDocument>, WhereConversionError>>()?,
+                        .collect::<Result<_, _>>()?,
                     operator: operator.try_into()?,
                 };
-                Ok(WhereDocument::WhereDocumentChildren(children))
+                Ok(Where::WhereChildren(children))
             }
             None => Err(WhereConversionError::InvalidWhere),
         }
     }
 }
 
-impl TryFrom<chroma_proto::WhereDocumentOperator> for WhereDocumentOperator {
+impl TryFrom<chroma_proto::WhereDocumentOperator> for DocumentOperator {
     type Error = WhereConversionError;
 
     fn try_from(proto_operator: chroma_proto::WhereDocumentOperator) -> Result<Self, Self::Error> {
         match proto_operator {
-            chroma_proto::WhereDocumentOperator::Contains => Ok(WhereDocumentOperator::Contains),
-            chroma_proto::WhereDocumentOperator::NotContains => {
-                Ok(WhereDocumentOperator::NotContains)
-            }
+            chroma_proto::WhereDocumentOperator::Contains => Ok(DocumentOperator::Contains),
+            chroma_proto::WhereDocumentOperator::NotContains => Ok(DocumentOperator::NotContains),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum SignedRoaringBitmap {
```

Comment:
I think some documentation around this type would be useful, as its not immediately clear what its used for and how. 

Also, I think its probably worth moving out into its own file.